Mathlib/Testing/SlimCheck/Sampleable.lean

```diff
@@ -275,15 +280,22 @@ instance sampleableExt [SampleableExt α] [Repr α] : SampleableExt (NoShrink α
 
 end NoShrink
 
-
 /--
 Print (at most) 10 samples of a given type to stdout for debugging.
 -/
--- Porting note: if `Control.ULiftable` is ported, make use of that here, as in mathlib3,
--- to enable sampling from higher types.
-def printSamples {t : Type} [Repr t] (g : Gen t) : IO PUnit := do
-  for i in List.range 10 do
-    IO.println s!"{repr (← g.run i)}"
+def printSamples {t : Type u} [Repr t] (g : Gen t) : IO PUnit :=
+  -- TODO: this should be a global instance
+  letI : MonadLift Id IO := ⟨fun f => pure <| Id.run f⟩
```

Comment:
I discounted https://github.com/leanprover/lean4/pull/5638, but maybe there's a use for it after all?

Reply:
The use here arises when working with monad stacks, and wanting to lift from `StateT S (ReaderT R Id)` to `StateT S (ReaderT R IO)` or similar.